src/NuGet.Core/NuGet.Build.Tasks/GetRestoreSolutionProjectsTask.cs

```diff
@@ -53,6 +53,11 @@ public override bool Execute()
                 }
 
                 var projectPath = Path.GetFullPath(Path.Combine(parentDirectory, project.ItemSpec));
+                
+                if (project.GetMetadata("BuildableProject").Equals("False"))
```

Comment:
@Forgind - Thanks for creating this pull request. I am curious if we are going to add a similar check on MSBuild side. The reason is even though restore succeeds build may fail because of lack of assets file.

Reply:
Ideally, MSBuild would create a metaproj for this with only a target to log a warning and not build the dwproj or other not buildable project.

I hadn't tested this sufficiently before making a PR; I tested it yesterday, and something still went wrong, so I reverted this to a draft. (I'm going on vacation tomorrow.)